### ISSUE TITLE
Regex not interpreted?

### ISSUE THREAD

**Issue by jrussell88** [FelisCatus/SwitchyOmega#442]
I would like to proxy only a small number of servers, for example 192.168.2.0/24

In order to achieve this I tried the following Regex expressions in the Bypass List: 
- (?!192\.168\.2\.[0-9]+) i.e. bypass all servers outside the range 192.168.2.0/24 therefore proxy only for the 192.168.2.0/24 range doesn't work
- 192\.168\.2\.[0-9]+ which would proxy everything but the range doesn't work
192\.168\.2\.1 bypasses the ip address

I also tested (?!www\.whatsmyip\.org)

Checking chrome://net-internals/#proxy it appears that the regex expressions are passed directly to Chrome without modification, however the Google page https://developer.chrome.com/extensions/match_patterns suggests wildcards and straightforward pattern matching. 

To what extent are Regex expressions implemented, or have I missed something? 

I'm running Chromium Version 41.0.2272.76 Ubuntu 14.04 (64-bit)

**Comment by FelisCatus** [comment]
These are not "match patterns." The help text in bypass list actually links to [the proxy bypass list format on Chrome extension docs](https://developer.chrome.com/extensions/proxy#bypass_list) page, where patterns are explained.

**tl;dr:** `192.168.2.0/24` is actually supported in bypass list without having to hand-craft regex for it. If you really need more regex power, consider using Switch Profile rules instead of bypass list.

**Comment by jrussell88** [comment]
Thanks for the advice, but as I'm trying to bypass every server except 192.168.2.0/24, putting it in the bypass list won't work. 

I've tried rules but that doesn't work either - perhaps I'm doing something wrong. My rules are:
Condition Type      Condition Details    Profile
URL Settings         192.168.2.*             Remote Lan
Default                                                  Direct

My understanding is that the first rule should filter 192.168.2.0/24 to the proxy, and urls which don't match the condition default to Direct. When I apply the rule, 192.168.2.0/24 doesn't go to the proxy and fails to connect; Other urls which should be direct eg whatsmyip.com go to the proxy.

Negating the first condition: !192.168.2.* doesn't change the results.

I'd appreciate if you can tell me what I'm doing wrong.

Thanks.

**Comment by FelisCatus** [comment]
Try selecting `Host wildcard` instead of `URL wildcard` in Condition Type. Also, make sure that there are no interfering lines in the bypass list.
